src/contexts/OracleDataContext.tsx

```diff
@@ -186,17 +185,59 @@ export function OracleDataProvider({ children }: { children: ReactNode }) {
 
   useEffect(() => {
     // its important this client only gets initialized once
-    Client([oraclesService, ...oracleEthersServices], {
+    Client([...oraclesServices, ...oracleEthersServices], {
       requests: (requests) => dispatch({ type: "requests", data: requests }),
       assertions: (assertions) =>
         dispatch({ type: "assertions", data: assertions }),
       errors: setErrors,
     });
   }, []);
 
-  errors.forEach((error, i) => {
-    if (error) console.warn("Oracle data error", i, errors);
-  });
+  useEffect(() => {
```

Comment:
this contains the fallback logic.  if any service errors, we check which service it is and react appropriately.
if its a graph service, we initiate a call to the web3 fallback service.
if its a web3 fallback service, we know that we are probably missing data, and need to show user link to legacy site